saltgui/static/scripts/api.js

```diff
@@ -76,7 +78,15 @@ class API {
   }
 
   _onRunReturn(command, data) {
-    const response = data.return[0];
+
+    // The data.return array may contain the answer from several batches
+    // combine these first
+    const response = { };
+    for(const batch of data.return) {
+      for(const host in batch) {
+        response[host] = batch[host];
+      }
+    }
```

Comment:
I suspect that this should not be in this pull request right?

Reply:
Traditionally, an array is returned with one value (one dictionary).
For batches, we know the array may contains more than one value (multiple dictionaries).
Collating the output in this way is generic. I cannot guarantee that all API calls will actually return an array of just one element. Since this was simple code and since it makes sure we do not accidentality lose a (partial) response, I did not delete it when I removed the support for batches.